_pytest/skipping.py

```diff
@@ -147,10 +148,17 @@ def getexplanation(self):
 
 @pytest.hookimpl(tryfirst=True)
 def pytest_runtest_setup(item):
-    evalskip = MarkEvaluator(item, 'skipif')
-    if evalskip.istrue():
-        item._evalskip = evalskip
-        pytest.skip(evalskip.getexplanation())
+    eval_skipif = MarkEvaluator(item, 'skipif')
+
+    if eval_skipif.istrue():
+        item._evalskip = eval_skipif
+        pytest.skip(eval_skipif.getexplanation())
+    elif isinstance(item.keywords.get('skip'), MarkInfo):
```

Comment:
Ah yes, sorry for not catching this. Will give it a closer look and see if there's a nice way to fix this. I'll also make sure my tests are updated to prevent this from happening again.